client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -208,7 +207,7 @@ public StreamInfo getStreamInfo(String scopeName, String streamName) {
     @Override
     public void close() {
         if (this.controller != null) {
-            this.controller.close();
+            Callbacks.invokeSafely(this.controller::close, ex -> log.error("Unable to close Controller client.", ex));
```

Comment:
Should all callers of `close()` use this `invokeSafely()` helper? Such as `KeyValueTableManagerImpl` above?

Reply:
Normally `close` shouldn't throw anything. But Java's `AutoCloseable.close` defines a `throws Exception` to handle the general case.

What I found is that the Controller and Controller client have some issues when closing. They use numerous sub-components and when they shut down, some may hang forever or throw an exception during shutdown. This prevents whatever else is supposed to close next to never be closed and stick in memory forever. 

The real fix is to figure out what is hanging or throwing exceptions and dealing with them appropriately. However the Controller is not my area of expertise, so I decided to leave that for another time (otherwise I would not be able to finish this by January).